File: 23. Implement strstr - GFG/23.-implement-strstr.cpp
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

int strstr(string ,string);

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        string a;
        string b;
        
        cin>>a;
        cin>>b;
        
        cout<<strstr(a,b)<<endl;
    }
}
// } Driver Code Ends


//Function to locate the occurrence of the string x in the string s.
int strstr(string s, string x){
     //Your code here
     int m = s.length();
     int n = x.length();
     for (int i = 0; i < m; i++){
         string y = s.substr(i,n);
         if(x==y){
             return i;
         }
     }
     return -1;
}

File: 23. Implement strstr - GFG/23.-implement-strstr.py
#Function to locate the occurrence of the string x in the string s.
def strstr(s,x):
    #code here
    k = len(x)
    for i in range(len(s)):
        if s[i:i+k] == x:
            return i
    return -1

#{ 
#  Driver Code Starts
#Contributed by : Nagendra Jha

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__ == '__main__':
    t=int(input())
    for cases in range(t):
        s,p =input().strip().split()
        print(strstr(s,p))


# } Driver Code Ends